src/DittoMachine.sol

```diff
@@ -186,38 +202,52 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
             _tokenId = FLOOR_ID;
         }
 
-        // calculate cloneId by hashing identifiying information
-        uint256 cloneId = uint256(keccak256(abi.encodePacked(
+        // calculate protoId by hashing identifiying information, precursor to cloneId
+        uint256 protoId = uint256(keccak256(abi.encodePacked(
             _ERC721Contract,
             _tokenId,
             _ERC20Contract,
             floor
         )));
+        // hash protoId and index to get cloneId
+        uint256 cloneId = uint256(keccak256(abi.encodePacked(protoId, index)));
 
-        _updatePrice(cloneId);
+        _updatePrice(protoId);
 
         uint256 value;
         uint256 subsidy;
+        uint256 elderId;
 
         if (ownerOf[cloneId] == address(0)) {
-
+            // check that index references have been set
+            if (protoIdToIndexToAfter[protoId][protoIdToIndexToPrior[protoId][index]] != index) {
+                // if references have not been set by a previous clone this clone cannot be minted
+                revert CloneNotFound();
+            }
             uint256 floorId = uint256(keccak256(abi.encodePacked(
                 _ERC721Contract,
                 FLOOR_ID,
                 _ERC20Contract,
                 true
             )));
+            floorId = uint256(keccak256(abi.encodePacked(floorId, index)));
 
             subsidy = _amount * MIN_FEE / DNOM; // with current constants subsidy <= _amount
             value = _amount - subsidy;
 
             if (cloneId != floorId && ownerOf[floorId] != address(0)) {
-                // check price of floor clone for price floor
+                // check price of floor clone to get price floor
                 uint256 minAmount = cloneIdToShape[floorId].worth;
                 if (value < minAmount) {
                     revert AmountInvalid();
                 }
             }
+            unchecked { if (index - protoIdToIndexHead[protoId] != 0) { // check cloneId at prior index
```

Comment:
isn't it equivalent to `index != protoIdToIndexHead[protoId]`? Also a new line after `unchecked {` for easier reading.

Reply:
yeah this will probably be better, i was using `depth` at one point for this and did a quick conversion. Will use the simpler check.